shared/src/commonMain/kotlin/com/example/tasky/dataSource/LoginDataSource.kt

```diff
@@ -36,4 +38,24 @@ class LoginDataSource(
             ResultWrapper.Error(DataError.Remote.UNKNOWN)
         }
     }
+
+    suspend fun register(registerBody: RegisterBody): ResultWrapper<Unit, BaseError> {
+        return try {
+            val response =
+                httpClient.post(Register()) {
+                    setBody(registerBody)
+                }
+
+            if (!response.isSuccess()) {
+                val errorResponse = response.body<ErrorResponse>()
+                println("register fail: ${errorResponse.message}")
+                return ResultWrapper.Error(DataError.Remote.UNKNOWN)
+            }
+
+            ResultWrapper.Success(Unit)
+        } catch (e: Exception) {
+            Throwable(e)
+            ResultWrapper.Error(DataError.Remote.UNKNOWN)
+        }
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `FUNCTION_NAME`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video Philipp shows the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc